### ISSUE TITLE
Make Topics collapsable or even better: dropdown menu

### ISSUE THREAD

**Issue by trueberryless** [HiDeoo/starlight-sidebar-topics#2]
### Is your feature request related to a problem?

Thinking about people who might use this fantastic plugin with many topics, the sidebar would always have many links on top which are rarely clicked because I think topics are not changed that often.

This takes up a lot of space which could be resolved in two ways, how one may ask...?

### Describe the solution you'd like

I can think of one rather easy to implement solution which would be nice already and another really cool solution (see below).

The first solution would make the topics collapsable, like Starlight sidebar groups can be collapsed. This would mean that if a user stays within one topic, they are neither distracted by other topics nor is the space in the sidebar wasted...

The maintainer could then choose if they want the topic section to be collapsable with an configuration option because this feature is only necessary if many topics exist.

### Describe alternatives you've considered

My second solution / thought would be really cool but harder to implement / it probably needs further discussion about exact design choices.

I was thinking about hiding the topic selection in a dropdown menu, so the user can select the topic once and then stays within the topic.

As an example for what I mean I can provide the following one: Shadcn/ui has a cleanly looking dropdown menu on this example page: https://ui.shadcn.com/examples/dashboard.
Look for the component that has the content "Alicia Koch" and click on it. That implementation is exactly what I envisioned for the topic selection for this plugin. But as I said, harder to implement and further discussion required...

The most ideal option imho would be that the maintainer can choose between all three modes (not collapsable, collapsable, dropdown menu). However, this is too good to be true (see below why)...

### Additional Context

Because I maintain one single Starlight plugin myself, I kinda know how hard it is to make users wishes / feature request come true. I have to admit that implementing ideas from other users doesn't make fun as much as implementing my own stuff cause most of the time I don't see the reasoning behind those ideas.

While I manage one plugin and struggle already, Hideoo manages -- how many are there now? -- like 10 or something. In other words: I don't know how they are able to achieve so many great things. And that's the reason why I think that all three selection options are too good to be true. It's the same problem like in so many other open-source projects. Instead of creating something together, people are just asking the maintainer to add new stuff and even more new stuff without contributing themselves. And because I don't like that, this feature request is not a request really. It's just an idea, not meant to be implemented anytime soon...

In short: don't make HiDeoo work too hard, he already did more than enough for the whole Astro community!

**Comment by HiDeoo** [comment]
Thanks for the detailed feedback.

> Thinking about people who might use this fantastic plugin with many topics, the sidebar would always have many links on top which are rarely clicked because I think topics are not changed that often.

My goal with this plugin is to actually avoid many topics. I personally think using that plugin should be done when having a large enough sidebar and where other attempts to reduce the number of entries have failed.

> The first solution would make the topics collapsable, like Starlight sidebar groups can be collapsed. This would mean that if a user stays within one topic, they are neither distracted by other topics nor is the space in the sidebar wasted...

If you end up using this plugin, I feel it's mandatory that all the topics are always visible at all time. How can you know you're even browsing the right topic if you can't see the other topics? Of course, this could be an option, but I think this option could lead to a lot of confusion.

> I was thinking about hiding the topic selection in a dropdown menu, so the user can select the topic once and then stays within the topic.

I'm not trying to hide topics and I think doing so would be a bad idea. Having the topics in a dropdown would require 1 extra click to even see the topics and an extra click to change topics. 

If a user prefers to have the topics in a dropdown, I think it would be better to use the [multi-sidebar plugin](https://starlight-utils.pages.dev/utilities/multi-sidebar/) and set the `switcherStyle` option to `dropdown`. Both plugins have their use cases with different behaviors and I don't think mixing them would be a good idea.

**Comment by trueberryless** [comment]
Okay, now I get the philosophy of the plugin. Thanks for the quick response!

**Comment by trueberryless** [comment]
For anyone stumbling across this issue and got interested in such a plugin, here you go: https://starlight-sidebar-topics-dropdown.trueberryless.org/

A huge thanks to @HiDeoo for creating the whole logic for this plugin, I just changed one file actually, so the credit goes to completely to him! <3
